pkg/image/builder.go

```diff
@@ -85,6 +85,9 @@ func GetPrivateImages(upstreamDir string, kotsKindsImages []string, checkedImage
 	objectsWithImages := make([]k8sdoc.K8sDoc, 0) // all objects where images are referenced from
 
 	for _, image := range kotsKindsImages {
+		if _, ok := checkedImages[image]; ok {
```

Comment:
How about moving this to the `else` part, so if `allPrivate` is set at a later point, all the flags get flipped.  Would happen if `proxyPublicImages` is set after initial install.